### PR TITLE
Pagination and swagger support were added

### PR DESCRIPTION
**Pull request by vgladkyi**

### CODE REVIEW

File: src/main/java/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -20,6 +23,7 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoints for managing books")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(value = "/api/books")
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

Add server.servlet.context-path=/api in application.properties

File: src/main/java/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -28,24 +32,31 @@ public class BookController {
     private final BookMapper bookMapper;
 
     @GetMapping
-    public List<BookDto> findAll() {
-        return bookService.findAll().stream()
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Display all available books", description = "Display all available books")
+    public List<BookDto> findAll(Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java]: Consider to add 
```suggestion
    public List<BookDto> findAll(@ParameterObject @PageableDefault Pageable pageable) {
```

File: src/main/java/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -54,14 +65,17 @@ public BookDto updateBook(@PathVariable Long id,
...
         bookService.deleteById(id);
     }
 
     @GetMapping("/search")
     @ResponseStatus(HttpStatus.OK)
-    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
-        return bookService.search(searchParameters)
+    @Operation(summary = "Display books by filter and with sorting",
+            description = "Display books by filter and with sorting")
+    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters, Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java]: The same

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's make it perfect

**Review by okuzan — APPROVED** [review]

**Pull request merged**